From f5b6c8cec8bb78dc118e31b55931005d782b3c51 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Fri, 4 Oct 2019 14:20:43 +0800
Subject: [PATCH] DRVLIB_SW-1183: Remove-cablibration-support-entry-from-dtsi

Removed intel,rx-auto-adapt flag from dtsi file and added the check in combophy driver to enable the calibration support
---
 drivers/phy/intel/phy-intel-combo.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/phy/intel/phy-intel-combo.c b/drivers/phy/intel/phy-intel-combo.c
index f9ab98dcc3a7..65a80d4b3757 100644
--- a/drivers/phy/intel/phy-intel-combo.c
+++ b/drivers/phy/intel/phy-intel-combo.c
@@ -77,11 +77,6 @@ enum intel_phy_mode {
 	PHY_MAX_MODE
 };
 
-struct intel_phy_calibrate {
-	u32 resv:31;
-	u32 rx_auto_adapt:1;
-};
-
 enum combo_phy_mode {
 	PCIE0_PCIE1_MODE = 0,
 	PCIE_DL_MODE, /* PCIe dual lane */
@@ -118,7 +113,6 @@ struct phy_ctx {
 	bool enable;
 	struct intel_combo_phy *parent;
 	enum intel_phy_mode phy_mode;
-	struct intel_phy_calibrate calibrate;
 	bool power_en;
 	enum intel_phy_role phy_role;
 	struct phy *phy;
@@ -658,7 +652,7 @@ static int intel_combo_phy_calibrate(struct phy *phy)
 	int ret = 0;
 
 	iphy = phy_get_drvdata(phy);
-	if (iphy->calibrate.rx_auto_adapt)
+	if (iphy->phy_mode == PHY_XPCS_MODE)
 		ret = phy_rxeq_autoadapt(phy);
 
 	return ret;
@@ -810,9 +804,6 @@ static int intel_phy_dt_parse(struct intel_combo_phy *priv,
 		}
 	}
 
-	iphy->calibrate.rx_auto_adapt =
-		device_property_read_bool(dev, "intel,rx-auto-adapt");
-
 	if (!(BIT(iphy->phy_mode) & priv->phy_cap)) {
 		dev_err(dev,
 			"PHY mode %u is not supported by COMBO PHY id %u of %s soc platform!\n",
